### ISSUE TITLE
Algorithm mto find Articulations points

### ISSUE THREAD

**Issue by ZigRazor** [ZigRazor/CXXGraph#235]
Please add an algorithm to find articulation points of a graph

**Comment by RachitGarg-12** [comment]
hey @ZigRazor  please assign me this issue

**Comment by ZigRazor** [comment]
Yes! @RachitGarg-12

**Comment by ZigRazor** [comment]
@RachitGarg-12 are you working on it?

**Comment by pradkrish** [comment]
is this still available? I am happy to work on this.

**Comment by ZigRazor** [comment]
Yes, is still available, I assign it to you @pradkrish !

**Comment by pradkrish** [comment]
Thanks. There seem to be two different approaches. A simple approach where vertices are removed one by one and see if removal of a vertex causes disconnected graph and another approach being Tarjan's algorithm. Which one do you prefer?

It's also possible to have both approaches, one can be called `articulation_points_simple()` and another one called `Tarjan()`.

**Comment by ZigRazor** [comment]
Taranto algorithm should be already implemented

**Comment by pradkrish** [comment]
Yes, [Tarjan's](https://github.com/ZigRazor/CXXGraph/issues/103) can be used to find both Strongly connected components and articulation points. In that case, do you think it's best to first work on Tarjan's and then use that implementation for finding articulation points?

**Comment by ZigRazor** [comment]
Yes, but I think for completness the best way is implement both algorithms

**Comment by pradkrish** [comment]
Not sure what you mean by _both_ here? :) Helpful if you can spell it out.

**Comment by ZigRazor** [comment]
Both for me means simple implementation and implementation with tarjan algorithm

**Comment by ZigRazor** [comment]
@pradkrish any news?

**Comment by ZigRazor** [comment]
De-assigned due to inactivity

**Comment by Sanjay007008** [comment]
Hai , I am Sanjay .

I came across this issue and would love to contribute an efficient algorithm for finding articulation points using Depth First Search (DFS). Before I start, I just wanted to check if this issue is still open.

I’m already familiar with the Bridges in Graph problem and DFS, so I’d love to hear if you have any specific expectations regarding implementation, coding style, or any other requirements. Happy to align my contribution accordingly!

**Comment by ZigRazor** [comment]
Hi, @Sanjay007008 this issue is still open, you can contributte following the coding style and the architecture of the project.
If you want I can assign this issue to you.
Thank you in advance

**Comment by Sanjay007008** [comment]
Hi @ZigRazor,

I'm excited to contribute to this issue! I have experience with C++ from problem-solving on LeetCode, but since I'm a beginner in open-source contributions, I want  any guidance on the coding style and architecture to follow. Please assign the issue to me so I can get started. Looking forward to your support!

You can reach me on Discord (sanjay7707) or via email at victorsanjay000@gmail.com for any further discussion.
How can I communicate with you?

**Comment by ZigRazor** [comment]
How can help you?
